Source/gravity/Gravity.cpp

```diff
@@ -1395,6 +1395,143 @@ Gravity::interpolate_monopole_grav(int level, RealVector& radial_grav, MultiFab&
     }
 }
 
+void
+Gravity::compute_radial_mass(const Box& bx,
+                             Array4<Real const> const u,
+                             RealVector& radial_mass,
+                             RealVector& radial_vol,
+                             int n1d, int level)
+{
+    const Geometry& geom = parent->Geom(level);
+
+    const auto dx     = geom.CellSizeArray();
+    const auto problo = geom.ProbLoArray();
+
+    Real dr = dx[0] / static_cast<Real>(gravity::drdxfac);
+    Real drinv = 1.0_rt / dr;
+
+    const int coord_type = geom.Coord();
+
+    AMREX_ALWAYS_ASSERT(coord_type >= 0 && coord_type <= 2);
+
+    GpuArray<Real, 3> center;
+    ca_get_center(center.begin());
+
+    Real octant_factor = 1.0_rt;
+
+    if (coord_type == 0) {
+
+        if ((std::abs(center[0] - problo[0]) < 1.e-2_rt * dx[0]) &&
+            (std::abs(center[1] - problo[1]) < 1.e-2_rt * dx[1]) &&
+            (std::abs(center[2] - problo[2]) < 1.e-2_rt * dx[2])) {
+
+            octant_factor = 8.0_rt;
+
+        }
+
+    } else if (coord_type == 1) {
+
+        if (std::abs(center[1] - problo[1]) < 1.e-2_rt * dx[1]) {
+
+            octant_factor = 2.0_rt;
+
+        }
+
+    }
+
+    Real fac = static_cast<Real>(gravity::drdxfac);
+
+    Real dx_frac = dx[0] / fac;
+
+    Real dy_frac = dx[1];
+    if (AMREX_SPACEDIM >= 2) {
+        dy_frac /= fac;
+    }
+
+    Real dz_frac = dx[2];
+    if (AMREX_SPACEDIM == 3) {
```

Comment:
isn't CellSizeArray() of size AMREX_SPACEDIM?  So you can't access dx[2] for < 3-d?

Reply:
Fixed